pool/src/lib.rs

```diff
@@ -767,6 +768,58 @@ impl LendingPool {
 
         Ok(())
     }
+
+    /// Calculate interest rate IR = MIN [ max_rate, base_rate / (1 - U)^alpha]
+    /// where
+    /// U - utilization, U = total_debt / total_collateral
+    /// alpha - parameter, by default 1.43 expressed as 143 with denominator 100
+    /// max_rate - maximal value of interest rate, by default 500% expressed as 50000
+    /// base_rate - base interest rate, by default 2%, expressed as 200
+    ///
+    /// For (1-U)^alpha calculation use binomial approximation with three terms
+    /// (1-U)^a = 1 - alpha * U + alpha/2 * (alpha - 1) * U^2 - alpha/6 * (alpha-1) * (alpha-2) * U^3
+    #[allow(dead_code)]
+    fn calc_interest_rate(
+        total_collateral: i128,
+        total_debt: i128,
+        alpha: u32,     // 143 / 100
+        max_rate: u32,  // 50000 / 10000
+        base_rate: u32, // 200 / 10000
+    ) -> Option<FixedI128> {
+        let u = FixedI128::from_rational(total_debt, total_collateral)?;
+        if u >= FixedI128::ONE {
+            return None; // utilization shouldn't be greater or equal one
+        }
+
+        let alpha = FixedI128::from_rational(alpha, ALPHA_DENOMINATOR)?;
+
+        let alpha_minus_one = alpha.sub(FixedI128::ONE)?;
+        let alpha_minus_two = alpha.sub(FixedI128::from_inner(FixedI128::DENOMINATOR * 2))?;
+
+        let u_pow_2 = u.mul(u)?;
```

Comment:
Would it be convenient to extend `FixedI128` with `pow` or `checked_pow` to replace it (and potentially other operations) with `u.pow(3)`?

Reply:
I think that done for efficiency - intermediate values also uses later

Reply:
Yeah, I see it. Just wondering if it's convenient to have `pow` in `FixedI128`.